### PR TITLE
Fixed deprecated _method errors

### PR DESCRIPTION
**Pull request by Padam87**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #1051 
| License       | MIT

### CODE REVIEW

File: Routing/Loader/Reader/RestActionReader.php [hunk 1]
```diff
@@ -200,13 +200,12 @@ public function read(RestRouteCollection $collection, \ReflectionMethod $method,
                 $condition    = null;
 
                 $annoRequirements = $annotation->getRequirements();
+                $annoMethods = $annotation->getMethods();
 
-                if (isset($annoRequirements['_method'])) {
```
Comment (stof) [inline Routing/Loader/Reader/RestActionReader.php]: there is another place using ``_method`` in this file

File: Tests/Fixtures/Etalon/annotated_conditional_controller.yml [hunk 2]
```diff
@@ -1,86 +1,91 @@
...
   path:      /users/{slug}/bget.{_format}
   controller:   ::bgetUserAction
-  requirements: {_method: GET}
+  method: GET
 
 bpost_user:
   path:      /users/{slug}/bpost.{_format}
   controller:   ::bpostUserAction
-  requirements: {_method: POST}
+  method: POST
```
Comment (stof) [inline Tests/Fixtures/Etalon/annotated_conditional_controller.yml]: this should be ``methods``, not ``method``

File: Tests/Routing/Loader/RestRouteLoaderTest.php [hunk 3]
```diff
@@ -101,10 +101,13 @@ public function testAnnotatedUsersFixture()
         foreach ($etalonRoutes as $name => $params) {
             $route = $collection->get($name);
 
-            if (!array_key_exists('_method', $route->getRequirements())) {
-                unset($params['requirements']['_method']);
+            if (!array_key_exists('requirements', $params)) {
+                $params['requirements'] = array();
```
Comment (stof) [inline Tests/Routing/Loader/RestRouteLoaderTest.php]: not needed actually

File: Tests/Routing/Loader/RestRouteLoaderTest.php [hunk 4]
```diff
@@ -101,10 +101,13 @@ public function testAnnotatedUsersFixture()
...
             }
 
+            // Symfony sets _method to keep BC, should be removed in 3.0
+            $params['requirements']['_method'] = $params['method'];
```
Comment (stof) [inline Tests/Routing/Loader/RestRouteLoaderTest.php]: It should ``methods``, not ``method``, to match the Symfony naming

File: Tests/Fixtures/Etalon/annotated_conditional_controller.yml (hunk 2 above)
Comment (Padam87) [inline Tests/Fixtures/Etalon/annotated_conditional_controller.yml]: Yep, I know, but the other test fixtures use it like this. I think fixing that should be another PR. Or should I change it here?

File: Routing/Loader/Reader/RestActionReader.php (hunk 1 above)
Comment (Padam87) [inline Routing/Loader/Reader/RestActionReader.php]: fixed

File: Routing/Loader/Reader/RestActionReader.php [hunk 5]
```diff
@@ -241,16 +239,11 @@ public function read(RestRouteCollection $collection, \ReflectionMethod $method,
...
-                unset($requirements['_method']);
-            } else {
-                $methods = array();
-            }
+            $methods = explode('|', strtoupper($httpMethod));
 
             // add route to collection
             $route = new Route(
-                $path, $defaults, $requirements, $options, $host, array(), $methods, $condition
+                $path, $defaults, array(), $options, $host, array(), $methods, $condition
```
Comment (stof) [inline Routing/Loader/Reader/RestActionReader.php]: this is wrong. Requirements should still be set in the route

Comment (Padam87) [inline Routing/Loader/Reader/RestActionReader.php]: At this point the `$requirements` only contained this:

https://github.com/FriendsOfSymfony/FOSRestBundle/pull/1054/files#diff-4f168cae49074c5f30b4abcccc85c281L186

Comment (stof) [inline Routing/Loader/Reader/RestActionReader.php]: not always. There is code setting the ``_format`` requirement a few lines above

**Comment by Padam87** [comment]
Also fixed the `method` -> `methods` problem.

**Comment by Padam87** [comment]
bump

**Comment by Padam87** [comment]
Awesome, thank you!

File: Controller/Annotations/Route.php [hunk 6]
```diff
@@ -22,9 +22,8 @@ class Route extends BaseRoute
     public function __construct(array $data)
     {
         parent::__construct($data);
-        $requirements = $this->getRequirements();
-        $requirements['_method'] = $this->getMethod();
-        $this->setRequirements($requirements);
+
+        $this->setMethods((array) $this->getMethod());
```
Comment (lsmith77) [inline Controller/Annotations/Route.php]: I guess this change is problematic, since we override the method, without checking if something was set before .. ?

Comment (lsmith77) [inline Controller/Annotations/Route.php]:
```
14:53 <Zerrvox> lsmith: FosRestBundle 1.7 has a BC break https://github.com/FriendsOfSymfony/FOSRestBundle/blob/bd233aca63b0965171d60fe4c539afa10d45633e/Controller/Annotations/Route.php
14:53 <lsmith> Zerrvox: what breaks?
14:55 <Zerrvox> If you have a route annotation in a FOSRest Controller the method of the generated route ends up being null
14:55 <Zerrvox> which cannot be loaded
```

Comment (Padam87) [inline Controller/Annotations/Route.php]: It shouldn't be a BC break.
`(array) null` ends up being an empty `[]`, which is is the default value for the methods property, and this code is in the constructor.

Althoug it might be a problem, tested it with both 1.6 and 1.7, and the methods part is definitely overwritten somewhere.

Comment (Padam87) [inline Controller/Annotations/Route.php]: https://github.com/FriendsOfSymfony/FOSRestBundle/pull/1067

Comment (lsmith77) [inline Controller/Annotations/Route.php]: fixed with https://github.com/FriendsOfSymfony/FOSRestBundle/releases/tag/1.7.1
